lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +251,34 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        url = ''
+        links = article_bs.find_all('a', hreflang='ru')
+        for link in links:
+            if 'Подробнее' in link.stripped_strings:
+                url = self.url_pattern + link.get('href')[len('/news')::]
+                if url not in self.urls:
+                    break
+        if url and url not in self.urls:
+            return url
+        return ''
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        seed_urls = self.get_search_urls()
+
+        for seed_url in seed_urls:
+            response = make_request(seed_url, self.config)
+            if not response.ok:
+                continue
+
+            article_bs = BeautifulSoup(response.text, "lxml")
+
+            for i in range(10):  # max at my site per seed page
```

Comment:
no, there should be no limit per page (you collect EVERYTHING from the page until total number of urls is collected)